src/testdir/test_search.vim

```diff
@@ -2194,5 +2194,13 @@ func Test_search_with_invalid_range()
   bwipe!
 endfunc
 
+func Test_searchstr()
+  new
+  call setline(1, ['  1', '  2 these', '  3 the', '  4 their', '  5 there', '  6 their', '  7 the', '  8 them', '  9 these', ' 10 foobar'])
+  call assert_equal("2 these\n  3", searchstr('\<2\s\+\_.*3\>'))
+
```

Comment:
Can you add test cases for passing invalid type of argument (for each argument) to searchstr() and check the error code?
Can you add a test case to search for a string that is not present?  Can you add a test case for searching for a multi-byte string?  Can you add test cases for a few more complex search patterns?  Can you add a test case to check whether the cursor has moved?